pages/tutorials/3.0/graphics/vertex-array.md

```diff
@@ -411,49 +435,46 @@ public:
             m_vertices[i].position += p.velocity * elapsed.asSeconds();
 
             // update the alpha (transparency) of the particle according to its lifetime
-            float ratio = p.lifetime.asSeconds() / m_lifetime.asSeconds();
+            float ratio           = p.lifetime.asSeconds() / m_lifetime.asSeconds();
             m_vertices[i].color.a = static_cast<std::uint8_t>(ratio * 255);
         }
     }
 
 private:
-
-    virtual void draw(sf::RenderTarget& target, sf::RenderStates states) const
+    void draw(sf::RenderTarget& target, sf::RenderStates states) const override
     {
         // apply the transform
         states.transform *= getTransform();
 
         // our particles don't use a texture
-        states.texture = NULL;
+        states.texture = nullptr;
 
         // draw the vertex array
         target.draw(m_vertices, states);
     }
 
-private:
-
     struct Particle
     {
         sf::Vector2f velocity;
-        sf::Time lifetime;
+        sf::Time     lifetime;
     };
 
     void resetParticle(std::size_t index)
     {
         // give a random velocity and lifetime to the particle
-        float angle = (std::rand() % 360) * 3.14f / 180.f;
-        float speed = (std::rand() % 50) + 50.f;
-        m_particles[index].velocity = sf::Vector2f(std::cos(angle) * speed, std::sin(angle) * speed);
+        const sf::Angle angle       = sf::degrees(std::rand() % 360);
+        const float     speed       = (std::rand() % 50) + 50.f;
+        m_particles[index].velocity = sf::Vector2f(speed, angle);
```

Comment:
Look at this usage of `sf::Vector2`'s new polar coordinate constructor. Much cleaner!